src/main/java/org/datadog/jenkins/plugins/datadog/listeners/DatadogGraphListener.java

```diff
@@ -103,13 +106,43 @@ public void onNewHead(FlowNode flowNode) {
             TagsUtil.addTagToTags(tags, "stage_depth", String.valueOf(stageDepth));
             // Add custom result tag
             TagsUtil.addTagToTags(tags, "result", result);
+            long pauseDuration = getPauseDuration(startNode);
+
             client.gauge("jenkins.job.stage_duration", getTime(startNode, endNode), hostname, tags);
+            client.gauge("jenkins.job.stage_pause_duration", pauseDuration, hostname, tags);
             client.incrementCounter("jenkins.job.stage_completed", hostname, tags);
         } catch (IOException | InterruptedException e) {
             DatadogUtilities.severe(logger, e, "Unable to submit the stage duration metric for " + getStageName(startNode));
         }
     }
 
+    private long getPauseDuration(@Nonnull FlowNode startNode) {
+        try {
+            long pauseDuration = 0;
+            FlowGraphWalker walker = new FlowGraphWalker(startNode.getExecution());
+
+            Iterator<FlowNode> it = walker.iterator();
+
+            // Iterate on the execution nodes to include pause duration of sub-stages
```

Comment:
Let's add a small comment to explain that part.

QQ: The FlowGraphWalker is instantiated while the pipeline is running (because called in onNewHead).
Are we sure this is safe to iterate through all nodes and are we sure we get the correct pause value?

Reply:
> QQ: The FlowGraphWalker is instantiated while the pipeline is running (because called in onNewHead).
Are we sure this is safe to iterate through all nodes and are we sure we get the correct pause value?

Even if the pipeline is running, nodes for which `startNode` is a parent are finished at this point. I added a condition to make sure we consider non-active nodes.